MaterialLib/Fluid/Density/WaterDensityIAPWSIF97Region1.h

```diff
@@ -0,0 +1,98 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   WaterDensityIAPWSIF97Region1.h
+ *
+ * Created on December 8, 2016, 4:19 PM
+ */
+
+#ifndef OGS_WATER_DENSITY_IAPWSIF97_REGION1_H
+#define OGS_WATER_DENSITY_IAPWSIF97_REGION1_H
+
+#include <string>
+
+#include "MaterialLib/Fluid/FluidProperty.h"
+#include "MaterialLib/Fluid/GibbsFreeEnergy/DimensionLessGibbsFreeEnergyRegion1.h"
+
+namespace MaterialLib
+{
+namespace Fluid
+{
+/** Water density model
+ *  base on the IAPWS Industrial Formulation 1997
+ *  <a href="http://www.iapws.org/relguide/IF97-Rev.pdf">IF97-Rev</a>
+ */
+class WaterDensityIAPWSIF97Region1 final : public FluidProperty
+{
+public:
+    WaterDensityIAPWSIF97Region1()
+        : _gibbs_free_energy(DimensionLessGibbsFreeEnergyRegion1()){};
+
+    /// Get density model name.
+    std::string getName() const override
+    {
+        return "Water density model based on IAPWSIF97";
+    }
+
+    /// Get density value.
+    /// \param var_vals Variable values in an array of temperature and pressure.
+    double getValue(const ArrayType& var_vals) const override
+    {
+        const double T = var_vals[static_cast<int>(PropertyVariableType::T)];
+        const double p = var_vals[static_cast<int>(PropertyVariableType::p)];
+        const double tau = _ref_T / T;
+        const double pi = p / _ref_p;
+
+        return p /
+               (_sR * T * (pi * _gibbs_free_energy.get_dgamma_dpi(tau, pi)));
```

Comment:
You can cancel out `p` and the `p` in the `pi` and save some of the parenthesis.